### PR TITLE
fix: check for prettier formatting in CI

### PR DESCRIPTION
**Pull request by estrattonbailey**
Fixes RNBW-4414

## What changed (plus any additional context for devs)
I was talking with Terry the other day and he mentioned that we used to fail CI if Prettier hadn't been run. Instead of using `eslint-plugin-prettier`, which integrates with ESLint and runs Prettier in a suboptimal way, I think we should go with [their recommended approach](https://prettier.io/docs/en/integrating-with-linters.html#notes) and run `prettier --check`.

This PR adds `format:check` script that runs in CI prior to the normal `lint` step.

Also, in #4100 I ran Prettier on everything so that we could remove the `--ext` flag in the near future. I figured I'd include that here as well.

## What to test
If CI passes on this PR we know all the code has been formatted correctly ✨ 

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [x] Did you test both iOS and Android?
- [x] If your changes are visual, did you check both the light and dark themes?
- [x] Added e2e tests? If not, please specify why
- [x] If you added new critical path files, did you update the CODEOWNERS file?
- [x] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: .github/workflows/macstadium-e2e.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
         run:  yarn audit-ci --moderate --config audit-ci.json
 
       - name: Lint
-        run: yarn lint
+        run: yarn format:check && yarn lint
```
Comment (terrysahaidak) [inline .github/workflows/macstadium-e2e.yml:49]: this should be part of `yarn lint` since it actually lints for mistakes

Comment (estrattonbailey) [reply .github/workflows/macstadium-e2e.yml:49]: Yeah good callout. I have [another PR open](https://github.com/rainbow-me/rainbow/pull/4070) to quiet logs in CI, which requires separate scripts. In general I think composing these scripts together is better than having one big script. Thoughts?

Comment (terrysahaidak) [reply .github/workflows/macstadium-e2e.yml:49]: and yarn lint checks for both ts and js linter errors
it used to be part of eslint, and people would call `yarn lint` locally to see what was falling

Comment (estrattonbailey) [reply .github/workflows/macstadium-e2e.yml]: Fair point. Done!

**Review by skylarbarrera — APPROVED** [review]
🏛️

**Review by terrysahaidak — APPROVED** [review]

**Review by perunt — APPROVED** [review]

**Review by brunobar79 — APPROVED** [review]

**Pull request merged**